plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -406,9 +406,10 @@ protected void createSchema(ConnectorSession session, Connection connection, Str
     }
 
     @Override
-    protected void dropSchema(ConnectorSession session, Connection connection, String remoteSchemaName)
+    protected void dropSchema(ConnectorSession session, Connection connection, String remoteSchemaName, boolean cascade)
             throws SQLException
     {
+        // ClickHouse always deletes all tables inside the database https://clickhouse.com/docs/en/sql-reference/statements/drop
```

Comment:
Do we want to prevent this if `cascade` was not specified? (and in MySQL, MariaDB and Singlestore too).

Reply:
Do you mean throwing an exception when `!cascade`? It means users always need to specify cascade option even when the target database is empty, right?

Reply:
What I was thinking of would require a list operation - and even then there can be races.
I don't know a good solution. My main concern is that if someone does a `DROP SCHEMA` (without `CASCADE`) they may not expect the operation to actually `CASCADE` to all objects in the schema.

Reply:
Added the test and manually tested without the same check in `DropSchemaTask`.